test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusUnitTest.java

```diff
@@ -381,6 +386,20 @@ public void afterEach(ExtensionContext context) throws Exception {
 
     @Override
     public void beforeEach(ExtensionContext context) throws Exception {
```

Comment:
Does this need to be done for each test method? Wouldn't it make sense to do it in `beforeAll`?

Reply:
`beforeAll` is not executed at all when a non-static field is used... that's the whole point of the issue ;-)

Reply:
OK, thanks

Reply:
Although I would have hoped there would be a more useful lifecycle method. Perhaps where the test class is constructed?

Reply:
`TestInstanceFactory.createTestInstance()` is not executed either.

Reply:
I rest my case then :)

Reply:
I like how @geoand asks the exact same questions I did xD
BTW this bit documents what is/isn't invoked if the registration is non-static https://junit.org/junit5/docs/current/user-guide/#extensions-registration-programmatic-instance-fields

Reply:
Great reference, thanks @manovotn !